tn_api_v2_core.yml

```diff
@@ -0,0 +1,956 @@
+                    brand:
+                        $ref: "#/components/schemas/FareInfoBrandDetails"
+
+        DatasourceBrandDetails:
+            type: array
+            description: List of branded fares for the current segment
+            items:
+                type: object
+                properties:
+                    base_price:
+                        description: Base price of the segment (tax excluded)
+                        type: float
+                        example: 0
+                    taxes:
+                        description: Tax price of the segment
+                        type: float
+                        example: 119.1
+                    fees:
+                        description: Fees for the segment *Travelport customers only*
+                        type: float
+                        example: 0
+                    price:
+                        description: Price of the segment (fare and tax included).
+                        type: float
+                        example: 197.1
+                    change_penalty:
+                        $ref: "#/components/schemas/ChangePenaltyWithExtraInfo"
+                    cancel_penalty:
+                        $ref: "#/components/schemas/CancelPenaltyWithExtraInfo"
+                    baggage_info:
+                        description: Contains baggage information.
+                        type: object
+                        properties:
+                            pieces:
+                                description: Number of baggages included
+                                type: int
+                                example: 1
+                            units:
+                                description: Unit of the baggage
+                                type: string
+                                example: "pc"
+                    fare_info:
+                        $ref: "#/components/schemas/BrandFareInfo"
+
+        BasicGenerateDatasourceRequest:
+            type: object
+            description: blah
+            required:
+                - segment_id
+                - total_price
+                - segment_source
+                - legs
+            properties:
+                segment_id:
+                    type: string
+                    description: This is the same as the segment_id in the segment_details response.
+                    example: "40495f0f293b54ffe64c833d987cd80ecdb5aca5"
+                total_price:
+                    type: float
+                    description: Total price of the segment (fare and tax included).
+                    example: 197.1
+                segment_source:
+                    type: string
+                    description: GDS or LCC provider source for segment. Only shown with ‘include_lcc’ request parameter enabled.
+                    example: travelport
+                legs:
+                    type: array
+                    description: A list of legs for this segment
```

Comment:
repeating above comment here aswell

Minor - I think this description could be improved a bit, as a list of legs is a little vague. Maybe

A list of required info for each leg of this segment.